docs/aws.md

```diff
@@ -0,0 +1,97 @@
+
+![Choose Instance details](../images/aws/launch-3.png)
+
+Add storage. The AWS EC2 launch wizard will suggest some form of storage by default. For our testing purposes this should be fine, but if you know that you need more storage, or a different storage performance requirements, then you can change the size and volume type here.
+
+![Choose Instance storage](../images/aws/launch-4.png)
+
+Add tags. It is recommended you add as many tags as you need to correctly identify this server later. Especially if you have many servers, it can be difficult to remember which one was used for which purpose. At a very minimum, add a `Name` tag with something memorable.
+
+![Add tags](../images/aws/launch-5.png)
+
+Configure security group. The security group is a stateful firewall that limits the traffic that is accepted by your instance. It is recommended to limit this as much as possible. In this case we are configuring it to only allow traffic on port 22 (ssh). It is recommended to restrict the source as well. To limit it to your workstation's current IP, click on the dropdown under "Source" and select "My IP".
+
+![Configure security group](../images/aws/launch-6.png)
+
+Launch instance. Click on "Launch", and select a key pair that you will use to connect to the instance. If you already have a keypair that you have used previously, you can reuse it here by selecting "Choose an existing key pair". Otherwise you can create a keypair now by selecting "Create a new key pair".
+
+![Launch instance](../images/aws/launch-7.png)
+
+Wait for instance to launch. When it is ready it will show as "running" under "Instance state", and "2/2 checks pass" under "Status Checks". Click on the instance to view the details on the bottom pane of the window, and look for the "IPv4 Public IP".
+
+![Wait for instance launch and view details](../images/aws/ec2-list.png)
+
+Connect to instance. Using the keypair that you used or created in the launch step and the IP in the previous step, run ssh. Be sure to use the `-A` option with ssh so that in a future step we will be able to use the same key to connect to a second instance.
+
+![Connect to instance](../images/aws/ssh-0.png)
+
+Run the following command in the SSH terminal. Note that there may be a more up to date version of the swift toolchain. Check https://swift.org/download/#releases for the latest available toolchain url for Amazon Linux 2.
+
+```
+SwiftToolchainUrl="https://swift.org/builds/swift-5.4.1-release/amazonlinux2/swift-5.4.1-RELEASE/swift-5.4.1-RELEASE-amazonlinux2.tar.gz"
+sudo yum install ruby binutils gcc git glibc-static gzip libbsd libcurl libedit libicu libsqlite libstdc++-static libuuid libxml2 tar tzdata ruby -y
+cd $(mktemp -d)
+wget ${SwiftToolchainUrl} -O swift.tar.gz
+gunzip < swift.tar.gz | sudo tar -C / -xv --strip-components 1
+```
+
+Finally, check that Swift is correctly installed by running the Swift REPL: `swift`.
+
+![Invoke REPL](../images/aws/repl.png)
```

Comment:
would it be easier to do the build stage in Docker? instead of downloading and installing the toolchain, one could build in docker on their development machine (or an EC2 instance for development/build/CI purposes), then scp to the runtime EC2 instance.

Reply:
🤷‍♂️  Both seem equally easy/hard to me. If you have a strong preference I can update it.

Reply:
@swift-server/sswg wdyt?

Reply:
dockering it up may be simpler for people to follow 👍 
A bit easier to point people at docker images than tell them what specific things to install every time.

Reply:
I’d just add another point of view - one might have a working ec2 setup but doesn’t use docker, which would make adding one more technology more complicated (my actual use case :-). That being said, either works of course, but just wanted to point out it adds another dimension of complexity.

Reply:
We could show both I guess? It’s not much more text to maintain tbh

Reply:
May view is that we should stick to compiling on the instance. Not everyone wants Docker and teaching people how to set up swift on a Linux box (even if it just points to the box) is not a bad thing.

IMO Docker should be a separate part of the guide and show the full lifecycle - compile container, push to ECR and deploy on to ECS/EKS etc

Reply:
This particular guide seems focused on stepping through someone's first experimentation with Server Side Swift. Like Tim said, there is value in a very beginner level guide and local or cross-instance compilation feels like it would just be introducing a complication earlier than needed.

Reply:
It seems like there's a couple of different opinions :-) i.e. either reduce the moving parts by compiling on the instance, or use docker to simplify the build command. I'm happy to go in either direction, just let me know :-)

Reply:
Friendly ping @tomerd ?

Reply:
how about we document both? this way folks that are familiar with Docker can go down that path, and others can install the toolchain on the host. hopefully we will have RPMs soon so the latter will be even simpler

Reply:
Yeah, I'm happy to do that. Give me a few days to wrap up another task on my plate and I'll get right on it.

Reply:
Just added a Docker option - I hope that's helpful

Reply:
Thanks a lot, that's great 👍